Add autoresponder to form block

arinaturcu opened the pull request:
<!-- Issues that this pull request closes. -->
Closes #.
<!-- Should look like this: `Closes #1, #2, #3.` . -->

### Summary
<!-- Please describe the changes you made. -->

### Screenshots <!-- if applicable -->

----

### Test instructions
<!-- Describe how this pull request can be tested. -->

<!--
#### Query
```javascript
new QueryQA().select('blocks').run()
```
-->

---- 

### Checklist before the final review

- [ ] Visual elements are not affected by independent changes.
- [ ] It is at least compatible with the [minimum WordPress version](https://wordpress.org/plugins/otter-blocks/).
- [ ] It loads additional script in frontend only if it is required.
- [ ] Does not impact the [Core Web Vitals](https://web.dev/vitals/).
- [ ] In case of deprecation, old blocks are safely migrated.
- [ ] It is usable in Widgets and FSE.
- [ ] Copy/Paste is working if the attributes are modified.

arinaturcu commented on inc/plugins/class-options-settings.php:
```diff
@@ -463,6 +480,297 @@ public function default_block() {
+	 * @access  public
+	 * @return array
+	 */
+	public static function get_allowed_mail_html() {
```
`wp_kses_post` allows some additional tags, for example, `form` or `input` which are not recommended in emails, that's why I added a different list. But do you think we should still use it?

HardeepAsrani replied:
I suppose you can get the wp_kses_allowed_html list for post and just filter out the form elements?

arinaturcu replied:
Done 👍🏻

HardeepAsrani commented:
Also, it doesn't need to be enabled by default and can make use of the Rich Editor that we use in Maps now that we allow HTML https://github.com/Codeinwp/otter-blocks/blob/865b9017d4be50f3d0af2dbab4e662548af3bfcb/src/blocks/blocks/leaflet-map/components/marker-editor.js#L4

<img width="253" alt="Screenshot 2023-03-17 at 3 32 18 AM" src="https://user-images.githubusercontent.com/2649903/225762700-ba5dfae0-60e6-437a-8945-a7e450b358d3.png">

arinaturcu commented:
Thanks, @HardeepAsrani!

I moved the Rich Text component from `leaflet-map` to `blocks/components`.
Also, I added the `allowRawHTML` prop to let users paste HTML inside.

https://github.com/Codeinwp/otter-blocks/blob/0751620f6a3af43eed260f46de16e3b1345d95dd/src/blocks/components/rich-text-editor/index.js#L23

So, if `allowRawHTML` is `true` the content will be decoded:
https://github.com/Codeinwp/otter-blocks/blob/0751620f6a3af43eed260f46de16e3b1345d95dd/src/blocks/components/rich-text-editor/index.js#L46-L48

Let me know it that's ok.

HardeepAsrani commented:
@arinaturcu I agree it's not an issue with Otter but if we're implementing this into Otter, we will need to figure out some way to make it work else replace it with something else, doesn't make much sense to add a RichText editor that doesn't work. Is it working nicely for Maps?

arinaturcu commented:
@HardeepAsrani, it's working the same for maps. I also checked with the current version of Otter and the problem is still there. It's not happening in the modal from the Google Map block though. Maybe there is an issue only between the rich text editor and the inspector sidebar.
That's why I was thinking of creating a separate issue to see how this can be solved as it seems to be a bigger issue.

HardeepAsrani commented:
@arinaturcu Okay, for now can we just make the editor load inside a modal as a quick fix with there being a button instead of RichText in the Editor?

HardeepAsrani commented:
@arinaturcu Yes, this is good enough for now.

rodica-andronache commented:
@arinaturcu the issues within the rich editor are gone now, but I'm not able to actually send the autoresponder mail anymore. 
Not sure if I'm missing something.

What I've tried: 
After I enable Autoresponder, I add some text for AUTORESPONDER SUBJECT and AUTORESPONDER BODY, and click on Publish. 
If I try to edit the post, I notice that the Autoresponder fields are missing.
https://www.loom.com/share/4fdf2353cea24673b763f7925f13ecdd

arinaturcu commented:
Thank you, @rodica-andronache! It should be fixed now.

rodica-andronache commented:
@arinaturcu thanks! didn't find anymore issues 👍
